tuning/data/data_processors.py

```diff
@@ -93,12 +93,36 @@ def load_dataset(
     def _process_dataset_configs(
         self, dataset_configs: List[DataSetConfig], **extra_kwargs
     ) -> Union[Dataset, IterableDataset]:
-        train_dataset = None
-        final_datasets = None
+
         splitName = "train"  # default
 
+        all_datasetdicts = []
+        sampling_probabilities = []
+
+        # quick check to see if we are sampling and if we need to throw error.
+        for d in dataset_configs:
+            if d.sampling:
+                sampling_probabilities.append(d.sampling)
+
+        if len(sampling_probabilities) > 0:
+            if len(sampling_probabilities) != len(dataset_configs):
+                raise ValueError(
+                    "Sampling probabilities should be provided for all datasets"
+                )
+            if sum(p for p in sampling_probabilities) != 1:
+                raise ValueError("Sampling probabilities don't sum to 1")
+            sample_datasets = True
+            logging.info(
+                "Sampling ratios are specified given datasets will be interleaved."
+            )
+        else:
+            logging.info(
+                "Sampling is not specified given datasets will be concatenated"
```

Comment:
If we choose to accept [this](https://github.com/foundation-model-stack/fms-hf-tuning/pull/408/files#r1876530327), then can make it as:

`Sampling is not specified; if multiple datasets are provided, the given datasets will be concatenated.`